src/azure-cli/azure/cli/command_modules/appconfig/_kv_helpers.py

```diff
@@ -0,0 +1,720 @@
+                continue
+            previous_kv = KeyValue(key, kv.value)
+
+            # No need to construct for properties format
+            if format_ == 'properties' or separator is None or not separator:
+                exported_dict.update({key: kv.value})
+                continue
+
+            key_segments = key.split(separator)
+            if key_segments[0].isdigit():
+                __export_keyvalue(key_segments,
+                                  kv.value, exported_list, key)
+
+            else:
+                __export_keyvalue(key_segments,
+                                  kv.value, exported_dict, key)
+
+        if exported_dict and exported_list:
+            logger.error("Can not export to a valid file! Some keys have been dropped. %s", json.dumps(
+                exported_dict, indent=2, ensure_ascii=False))
+
+        return __compact_key_values(exported_dict if not exported_list else exported_list)
+    except Exception as exception:
+        raise CLIError("Fail to export key-values." + str(exception))
+
+
+def __flatten_dict_to_properties(value_to_flatten, separator='.', prefix=''):
+    flattened_properties_dict = {}
+    if isinstance(value_to_flatten, dict):
+        for keys, values in value_to_flatten.items():
+            current_prefix = prefix + separator + str(keys) if prefix else str(keys)
+            if isinstance(values, (dict, list)):
+                flattened_properties_dict.update(__flatten_dict_to_properties(values, separator, current_prefix))
+            else:
+                flattened_properties_dict.update({current_prefix: str(values)})
+
+    elif isinstance(value_to_flatten, list):
+        for idx, val in enumerate(value_to_flatten):
+            current_prefix = prefix + '[' + str(idx) + ']' if prefix else str(idx)
+            if isinstance(val, (dict, list)):
+                flattened_properties_dict.update(__flatten_dict_to_properties(val, separator, current_prefix))
+            else:
+                flattened_properties_dict.update({current_prefix: str(val)})
+    else:
+        flattened_properties_dict.update({prefix: str(value_to_flatten)})
+
+    return flattened_properties_dict
+
+
+def __export_features(retrieved_features, format_):
+    exported_dict = {}
+    client_filters = []
+    if format_ == 'json':
+        exported_dict["FeatureManagement"] = {}
+
+    elif format_ in ('yaml', 'properties'):
+        # Currently, this condition will never be met
+        # We only support json feature flags for now
+        featureSetValues = {}
+        featureSetValues["features"] = {}
+        exported_dict["feature-management"] = {"featureSet": featureSetValues}
+
+    try:
+        # retrieved_features is a list of FeatureFlag objects
+        for feature in retrieved_features:
+
+            # if feature state is on or off, it means there are no filters
+            if feature.state == "on":
+                feature_state = True
```

Comment:
feature.state is always lower case? either we convert to lower or feature.state can be an enum.

Reply:
FeatureFlag object has state attribute whose value is always converted to lowercase.

---
In reply to: [356324912](https://github.com/Azure/azure-cli/pull/11467#discussion_r356324912) [](ancestors = 356324912)